### DISCUSSION TITLE
filtering liveness probe access logs

### DISCUSSION THREAD

**Discussion by glehmann** [static-web-server/static-web-server#236] [Q&A] [answered]
we are using sws in kubernetes, with a liveness probe configured to check the status of the sws instance

~~~yaml
          livenessProbe:
            httpGet:
              path: /
              port: http
            periodSeconds: 5
~~~

this works very well, but generates a lot of logs of very low value for us:

~~~
…
2023-07-06T09:29:08.148155Z  INFO static_web_server::handler: incoming request: method=GET uri=/
2023-07-06T09:29:13.148495Z  INFO static_web_server::handler: incoming request: method=GET uri=/
2023-07-06T09:29:18.148100Z  INFO static_web_server::handler: incoming request: method=GET uri=/
2023-07-06T09:29:23.147758Z  INFO static_web_server::handler: incoming request: method=GET uri=/
…
~~~

I've found a way to globally change the log level, but that would also disable the logs for the actual requests.

Is there a way to filter the logs in sws?

**Comment by joseluisq** [comment]
Unfortunately, there is no way from the settings right now to keep some access `info` level logs and disable others. E.g If you go with `error` level then `warn`, `info` and all the verbose ones will be not shown. 

So probably we will need to introduce some way for fine-grained logs. Probably via some _access log_ setting or so. See [discussion](https://github.com/static-web-server/static-web-server/discussions/206).
If you have ideas feel free to let me know.

**Reply by joseluisq** [reply]
Another thing that I'm thinking of is a built-in `GET /health` endpoint to health check SWS but just a random idea.

**Reply by glehmann** [reply]
I like your idea of a builtin `/health` endpoint. Do you think that may not be to difficult to implement? If no I may give it a try :)

**Reply by joseluisq** [reply]
Then you could give our codebase a try.
I think you could start proposing this in an issue.

**Comment by joseluisq** [comment]
A question, is your liveness probe also caring about the logs shown, or is it just enough with receiving a `200 OK`?

Additionally, if the liveness probe cares only about receiving a successful response then you could try to send a `HEAD` request instead.
See [HTTP Methods Supported](https://static-web-server.net/features/http-methods/#http-methods-supported)

**Reply by glehmann** [reply]
That would have been really nice, but unfortunately k8s does'nt support HEAD http probes yet. There are some discussions in https://github.com/kubernetes/enhancements/pull/4012 but I have no idea if it will ever make it to the stable version.

**Comment by joseluisq** [comment, marked as answer]
This is resolved by https://github.com/static-web-server/static-web-server/pull/238
